File: Ejemplos/43-DockerAndTestContainers/src/main/java/dev/joseluisgs/services/PersonasService.java
package dev.joseluisgs.services;

import dev.joseluisgs.errors.PersonaError;
import dev.joseluisgs.model.Persona;
import io.vavr.control.Either;

import java.util.List;

public interface PersonasService {
    Either<PersonaError, List<Persona>> getAll();

    Either<PersonaError, Persona> getById(int id);

    Either<PersonaError, Persona> create(Persona persona);

    Either<PersonaError, Persona> update(long id, Persona persona);

    Either<PersonaError, Persona> delete(long id);
}

File: Ejemplos/33-EjemplosTestRepositorioServicio/src/main/kotlin/personas/services/PersonasService.kt
package dev.joseluisgs.personas.services

import dev.joseluisgs.personas.model.Persona
import java.util.*

interface PersonasService {
    fun getAll(): List<Persona>
    fun findByName(name: String): List<Persona>
    fun getById(id: UUID): Persona
    fun save(persona: Persona): Persona
    fun update(id: UUID, persona: Persona): Persona
    fun delete(id: UUID): Persona
}